### ISSUE TITLE
I would like a mentor

### ISSUE THREAD

**Issue by Elentary** [argoproj/argo-workflows#10843]
**Question: Have you read the [mentoring guide](https://argoproj.github.io/argo-workflows/mentoring/)? Please only submit this request for general mentoring (not GSoC).**

Answer: **Yes I have read the mentoring guide.**

**What is your background? Any experience with Go, Kubernetes, React, Typescript, etc.?**

Answer: **I have decent experience in C++, Python and programming in general. Decent proficiency in Kubernetes and several months of Argo operations & maintaining.**

**Is there any particular issue you'd like to work on? You may want to check out the [list of good first issues](https://github.com/argoproj/argo-workflows/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).**

Answer: **I would like to support virtual-hosted-style for S3 artefacts and I have already made a local hot fix but currently I am stuck with errors with codegen**

**Do you want to work with your mentor async (i.e. by Slack messaging or issue comments), or sync (e.g. using Zoom video)?** Please include your time-zone if you want to work sync.

Answer: async (Slack is good)

**Comment by terrytangyuan** [comment]
> but currently I am stuck with errors with codegen

What is the error you are seeing?

**Comment by Elentary** [comment]
@terrytangyuan Finally I was able to do a successful codegen inside devcontainer but **only as root user** and with tweaks I had to do.

My current working setup consisted of doing the following steps :
1. Change `devcontainer.json` in the way so it would mount workspace not to `/home/vscode/go` but rather to `/go`, as `GOPATH` env is `/go` inside a container and Argo code generation requires workspace path to be `$GOPATH/src/github.com/argoproj/argo-workflows`.
```
   "onCreateCommand": ".devcontainer/pre-build.sh",
-  "workspaceMount": "source=${localWorkspaceFolder},target=/home/vscode/go/src/github.com/argoproj/argo-workflows,type=bind",
-  "workspaceFolder": "/home/vscode/go/src/github.com/argoproj/argo-workflows"
+  "workspaceMount": "source=${localWorkspaceFolder},target=/go/src/github.com/argoproj/argo-workflows,type=bind",
+  "workspaceFolder": "/go/src/github.com/argoproj/argo-workflows"
 }
```

2. Login as `root` (`sudo su`) because otherwise `make codegen` would fail being ran as `vscode` user:
`fatal: could not create leading directories of '/go/src/github.com/gogo/protobuf': Permission denied`
P.S. Do not forget to generate .kubeconfig again for `root` user

3. Install missing `protoc` as it is not being installed in [pre-build script](https://github.com/argoproj/argo-workflows/blob/f418eacddc7e6a95b5793180a2c43f7408d3c0f9/.devcontainer/pre-build.sh) - `apt install protobuf-compiler`:
`2023/04/07 18:02:52 Unable to find 'protoc': exec: "protoc": executable file not found in $PATH`

These 3 steps are enough to do a successful codegen on master (e734ae524). However I am 100% sure that my way is not the correct one, because it creates lots of problems after:
1. `root` ownership of new files and folders when VSCode does all its operations as `vscode` user and fails because of permissions.
2. Warnings from webpack, because there are missing src files in node_modules. I don't know why there are not being generated, `ui` and `ui-deps` steps are successful. These warnings does not prevent me from using Argo, but very annoying
```
WARNING
./node_modules/autolinker/dist/es2015/anchor-tag-builder.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/go/src/github.com/argoproj/argo-workflows/ui/node_modules/autolinker/src/anchor-tag-builder.ts' file: Error: ENOENT: no such file or directory, open '/go/src/github.com/argoproj/argo-workflows/ui/node_modules/autolinker/src/anchor-tag-builder.ts'
```

Also I noticed that `kit up` does not rebuild binaries when I change the code 🤯. This is completely unexpected behaviour!
I changed `workflow/artifacts/s3/s3.go` file, ran `kit up` and Argo was launched from old binary missing all my changes.
I was super-surprised, because I see in terminal it reacts to a file change and relaunches corresponding steps (e.g. build-argo and server).
So currently I manually remove `dist` folder after each change of code so it would 100% use new binaries.

TL;DR Summary
I would like a help from an experienced Argo developer to a make a proper workspace setup. I guess using `root` user is not a good solution and will result with more surprises for me in the future.
Also I need a person whom I can ask questions from time to time that answers to which are missing in the docs. And what part of them will be cool to make an issue from or even a PR with docs or configuration code improvements.

**Comment by Elentary** [comment]
My organisation uses Argo Workflows heavily for computation and processing and I am the guy responsible for Argo.
We have already discovered a number of bugs or unsupported stuff that we actually need, so I and my colleagues would be happy to communicate, learn the code and contribute more to Argo Workflows.

**Comment by terrytangyuan** [comment]
Feel free to message me on Slack and I'll try help you get started.

**Comment by terrytangyuan** [comment]
Let’s discuss on Slack

**Comment by GeunSam2** [comment]
i have same issue about `autolinker` warning @Elentary 
have you any solution about this warning?
```
WARNING
./node_modules/autolinker/dist/es2015/anchor-tag-builder.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/go/src/github.com/argoproj/argo-workflows/ui/node_modules/autolinker/src/anchor-tag-builder.ts' file: Error: ENOENT: no such file or directory, open '/go/src/github.com/argoproj/argo-workflows/ui/node_modules/autolinker/src/anchor-tag-builder.ts'
```
